R/narr_data_functions.R

```diff
@@ -1,50 +1,27 @@
 #' @import data.table
 
-read_narr_fst_join <- function(d_one, narr_variables, narr_fst_filepath) {
-  d_orig <- d_one
-  d_orig$row_index <- seq_len(nrow(d_orig))
-
-  narr_cell <- unique(d_one$narr_cell)
-  narr_row_start <- ((narr_cell - 1) * 7671) + 1
-  narr_row_end <- narr_cell * 7671
-
-  d_one <-
-    d_one %>%
-    dplyr::mutate(row_index = seq_len(nrow(d_one))) %>%
-    dplyr::group_by(row_index) %>%
-    tidyr::nest() %>%
-    dplyr::mutate(date_seq = purrr::map(data, ~ seq.Date(.$start_date, .$end_date, by = 1))) %>%
-    tidyr::unnest(cols = c(data, date_seq)) %>%
-    dplyr::select(row_index, narr_cell, date = date_seq)
-
-  out <-
-    fst::read_fst(
-      path = narr_fst_filepath,
-      from = narr_row_start,
-      to = narr_row_end,
-      columns = c("narr_cell", "date", narr_variables),
-      as.data.table = TRUE
-    )
-
-  # subset to all dates needed for the narr cell number
-  out <- out[list(data.table::CJ(narr_cell, unique(d_one$date))), nomatch = 0L]
-
-  d_one <- dplyr::left_join(d_one, out, by = c("date", "narr_cell"))
-
-  d_out <-
-    d_one %>%
-    dplyr::group_by(row_index) %>%
-    dplyr::summarize_at(tidyselect::all_of(narr_variables), mean, na.rm = TRUE)
+read_join_chunks <- function(d_one, narr_chunk_path) {
+  message('reading in ', narr_chunk_path, ' and joining to data...')
+  narr_chunk <- fst::read_fst(narr_chunk_path, as.data.table = TRUE)
+  d_out <- merge(data.table::as.data.table(d_one), narr_chunk, by = c("narr_cell", "date"))
+  remove(narr_chunk)
+  return(d_out)
+}
 
-  dplyr::left_join(d_orig, d_out, by = "row_index") %>%
-    dplyr::select(-row_index)
+download_join_chunks <- function(d_one, narr_product) {
+  narr_chunk <- unique(d_one$narr_chunk)
+  message('downloading narr fst chunk files for chunk number ', narr_chunk, '...')
```

Comment:
This message might not be necessary, because we are getting built in messages from `s3::s3_get_files()`, right?

It seems useful to be able to communicate how many different chunks will need to be downloaded and processed, and then to update the user as they complete.  Could the `s3` messages and confirmation prompts be used for this purpose?  Or would we need to implement it here?